sdk/src/main/java/com/mapbox/maps/StyleObserver.kt

```diff
@@ -60,18 +63,29 @@ internal class StyleObserver(
    * Invoked when a style has loaded
    */
   override fun onStyleLoaded(eventData: StyleLoadedEventData) {
-    val style = Style(nativeMap, pixelRatio)
-    loadedStyle?.markInvalid()
-    loadedStyle = style
-    styleLoadedListener.onStyleLoaded(style)
+    invokeStyleLoadedListeners()
+  }
 
-    loadStyleListener?.onStyleLoaded(style)
-    loadStyleListener = null
+  private fun invokeStyleLoadedListeners() {
+    loadedStyle?.let { style ->
+      styleLoadedListener.onStyleLoaded(style)
+
+      loadStyleListener?.onStyleLoaded(style)
+      loadStyleListener = null
 
-    getStyleListeners.forEach { listener ->
-      listener.onStyleLoaded(style)
+      getStyleListeners.forEach { listener ->
+        listener.onStyleLoaded(style)
+      }
+      getStyleListeners.clear()
     }
-    getStyleListeners.clear()
+  }
+
+  private fun invokeStylePreloadedListeners() {
+    val style = Style(nativeMap, pixelRatio)
+    loadedStyle?.markInvalid()
+    loadedStyle = style
```

Comment:
shouldn't that logically be located in `invokeStyleLoadedListeners()`? although that events (style preloaded and style loaded) are close enough - guess it makes sense to mark previous style invalid and store new style in a var in `invokeStyleLoadedListeners()`

Reply:
This is the place we save the reference of the newly created Style object inside the preload phase, and we are returning the same object inside the `OnStyleLoaded` callback. 

Imo we should mark the previous style invalid at this point of time, as the new Style object is already created, previous style url is already overwritten.